src/contrib/mlir/compiler/dialect/ops_interfaces.td

```diff
@@ -0,0 +1,64 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+//
+// This is the nGraph Dialect operation interfaces definitions
+//
+//===----------------------------------------------------------------------===//
+
+#ifdef NG_OP_INTERFACES
+#else
+#define NG_OP_INTERFACES
+
+#ifdef OP_BASE
+#else
+include "mlir/IR/OpBase.td"
+#endif // OP_BASE
+
+// Op Interfaces for Op Versions
+// They are empty for now. To check the version of an op, we do:
+// 	Operation *op = …;
```

Comment:
Only up to 9. Then you need a regexp. If `Foo` the same as `Foo0` and `Foo000`? I think you just need to define version syntax and say "Don't to that" for variant ways to write the number since they'd be different classes.